### ISSUE TITLE
[0.3 release test] java.lang.ClassNotFoundException: com.intel.analytics.bigdl.nn.Scheduler$Ready in Scala rnn test

### ISSUE THREAD

**Issue by hkvision** [intel-analytics/BigDL#1745]
https://github.com/intel-analytics/BigDL/tree/branch-0.3/spark/dl/src/main/scala/com/intel/analytics/bigdl/models/rnn
After I do the training locally,
when I do the test using a saved model with the following command:
`spark-submit \
--master local[*] \
--executor-cores 4 \
--total-executor-cores 4 \
--class com.intel.analytics.bigdl.models.rnn.Test \
dist/lib/bigdl-0.3.0-SNAPSHOT-jar-with-dependencies.jar \
-f /home/kai/Downloads/rnn/saveDict --model /home/kai/Downloads/rnn/model/20171025_141505/model.13706 --words 20`

I got the below exception:
17/10/25 16:22:22 INFO Engine$: Auto detect executor number and executor cores number
17/10/25 16:22:22 INFO Engine$: Executor number is 1 and executor cores number is 4
17/10/25 16:22:22 INFO Engine$: Find existing spark context. Checking the spark conf...
Exception in thread "main" java.lang.ClassNotFoundException: com.intel.analytics.bigdl.nn.Scheduler$Ready
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at java.io.ObjectInputStream.resolveClass(ObjectInputStream.java:677)
	at java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:1826)
	at java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1713)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2000)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1535)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:422)
	at scala.collection.mutable.HashMap$$anonfun$readObject$1.apply(HashMap.scala:143)
	at scala.collection.mutable.HashMap$$anonfun$readObject$1.apply(HashMap.scala:143)
	at scala.collection.mutable.HashTable$class.init(HashTable.scala:106)
	at scala.collection.mutable.HashMap.init(HashMap.scala:40)
	at scala.collection.mutable.HashMap.readObject(HashMap.scala:143)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at java.io.ObjectStreamClass.invokeReadObject(ObjectStreamClass.java:1058)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2136)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2027)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1535)
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2245)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2169)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2027)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1535)
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2245)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2169)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2027)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1535)
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2245)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2169)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2027)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1535)
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2245)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2169)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2027)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1535)
	at java.io.ObjectInputStream.readArray(ObjectInputStream.java:1933)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1529)
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2245)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2169)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2027)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1535)
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2245)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2169)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2027)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1535)
	at java.io.ObjectInputStream.readArray(ObjectInputStream.java:1933)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1529)
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2245)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2169)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2027)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1535)
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2245)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2169)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2027)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1535)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:422)
	at com.intel.analytics.bigdl.utils.File$.load(File.scala:170)
	at com.intel.analytics.bigdl.nn.Module$.load(Module.scala:44)
	at com.intel.analytics.bigdl.models.rnn.Test$$anonfun$main$1.apply(Test.scala:51)
	at com.intel.analytics.bigdl.models.rnn.Test$$anonfun$main$1.apply(Test.scala:42)
	at scala.Option.foreach(Option.scala:257)
	at com.intel.analytics.bigdl.models.rnn.Test$.main(Test.scala:42)
	at com.intel.analytics.bigdl.models.rnn.Test.main(Test.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.SparkSubmit$.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:755)
	at org.apache.spark.deploy.SparkSubmit$.doRunMain$1(SparkSubmit.scala:180)
	at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:205)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:119)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)

Anywhere I'm doing wrong?

**Comment by zhangxiaoli73** [comment]
Please change "-f /home/kai/Downloads/rnn/saveDict" to "-f /home/kai/Downloads/rnn/test", and try again

**Comment by hkvision** [comment]
Seems that there is problem in the jar when loading the model?

**Comment by yiheng** [comment]
add a --driver-class-path to the spark submit option

**Comment by zhangxiaoli73** [comment]
yeah. add --driver-class-path is ok now.

**Comment by hkvision** [comment]
After adding `--driver-class-path dist/lib/bigdl-0.3.0-SNAPSHOT-jar-with-dependencies.jar \`, the problem solved.
We need to update the README file.

**Comment by hkvision** [comment]
Some minor modification that makes the readme consistent and easier to follow:

1. In `Test the model`
Please create a test.txt file under the folder in which you save your dictionary during training process.
->
Please create a test.txt file under the folder `/path/saveDict`in which you save your dictionary during training process.

2. In `Test the model` example command:
-f /dictDirect  -> `-f /path/saveDict`
--model /modeldirectory/model.iterationNumber -> `--model /path/model/model.iterationNumber`
